packages/concerto-core/lib/introspect/introspector.js

```diff
@@ -59,11 +59,12 @@ class Introspector {
         const modelFiles = this.modelManager.getModelFiles();
         for(let n=0; n < modelFiles.length; n++) {
             const modelFile = modelFiles[n];
-            result = result.concat(
-                modelFile.getAllDeclarations()
-                    .filter(declaration => !declaration.isScalarDeclaration?.()
-                    )
-            );
+
+            const filteredDeclarations = modelFile.getAllDeclarations()
+                .filter(declaration => !declaration.isScalarDeclaration?.())
+                .filter(declaration =>  !declaration.isMapDeclaration?.());
```

Comment:
Combining the conditions to avoid another loop

```suggestion
                .filter(declaration => !declaration.isScalarDeclaration?.() && !declaration.isMapDeclaration?.());
```